src/metabase/models/task_history.clj

```diff
@@ -46,13 +46,37 @@
       (perms/application-perms-path :monitoring)
       "/")})
 
+(defn- task-details-for-snowplow
+  "Ensure task_details is less than 2048 characters.
+
+  Most of the times it's < 200 characters, but there are cases task-details contains an exception.
+  In those case, we want to make sure the stacktrace are ignored from the task-details."
+  [task-details]
+  (let [;; task-details is {:throwable e} during sync
+        ;; check [[metabase.sync.util/run-step-with-metadata]]
+        task-details (cond-> task-details
+                       (some? (:throwable task-details))
+                       (update :throwable dissoc :trace :via)
+
+                       ;; if task-history is created via `with-task-history
+                       ;; the exception is manually caught and includes a stacktrace
+                       true
+                       (dissoc :stacktrace)
+
+                       true
+                       (dissoc :trace :via))
+        as-string     (json/generate-string task-details)]
+    (if (>= (count as-string) 2048)
+      ""
```

Comment:
I would have expected the result to be `nil` to indicate the absence of a value, not an empty string. It doesn't change the runtime behaviour at all, but it's arguably more idiomatic?

And this conditional is missing from the docstring. The stacktrace is already being removed, so is there another way this code path can be reached? Or is this just for extra safety, in case there's another way the task_details can reach 2048 characters?

Reply:
> "" to nil

agreed.


> update docs

Will do, I certainly missed this!

> extra safety?

yes, the above conditions covered all cases we could have an exception in our code base as of now.
As I checked on our stats, most task_details without exception are < 200 chars.

but I think there is always good to have a safe guard here. But I'll add a log message just in case?